[import-w3c-tests] Honor skip expectation with --src-dir

gsnedders opened the pull request:
#### e21f2280dbbbfe4fcc4801dcf4ad83d09f0621d8
<pre>
[import-w3c-tests] Honor skip expectation with --src-dir
<a href="https://bugs.webkit.org/show_bug.cgi?id=257198">https://bugs.webkit.org/show_bug.cgi?id=257198</a>
rdar://problem/110004053

Reviewed by Jonathan Bedard.

Previously, we only applied the skip logic in test_downloader, and hence
it only had any effect when downloading tests, and not when importing
from a given --src-dir.

* Tools/Scripts/webkitpy/w3c/test_downloader.py:
(TestDownloader._add_test_suite_paths):
(TestDownloader.copy_tests.should_copy_dir):
(TestDownloader.copy_tests.should_copy_file):
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.should_skip_path):
(TestImporter.find_importable_tests):
(TestImporter.should_skip_file): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: <a href="https://commits.webkit.org/267088@main">https://commits.webkit.org/267088@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3e60ef94b5aaefe53ae20116c5e3a248125b343d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15467 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15771 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16134 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17221 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14502 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15632 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18287 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15867 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17078 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15651 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16081 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13150 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17964 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/15574 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13352 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13950 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20883 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14413 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14117 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17384 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14697 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12451 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13955 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18316 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1903 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14518 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -264,12 +270,25 @@ def test_downloader(self):
+            if parent in paths_to_skip:
+                return True
+
+            if parent in paths_to_import:
```
No entirely clear what the point of `paths_to_import` is if it's checked after `paths_to_skip`, and the default is `False`. I would expect `paths_to_import` to be checked first.

gsnedders replied:
Consider:

```json
{
  "a": "import",
  "a/b": "skip",
  "a/b/c": "import"
}
```

When processing `a/b/c`, we need to exit the loop early when we reach the import.

The ordering doesn't matter, because in theory they're disjoint sets. Could add an assert that they're disjoint to clarify that.

JonWBedard replied:
Ah! This example clears up the difference.

gsnedders replied:
Note `test_skip_test_import` already covers this—so getting this wrong would've regressed the tests!

gsnedders committed:
[import-w3c-tests] Honor skip expectation with --src-dir
https://bugs.webkit.org/show_bug.cgi?id=257198
rdar://problem/110004053

Reviewed by Jonathan Bedard.

Previously, we only applied the skip logic in test_downloader, and hence
it only had any effect when downloading tests, and not when importing
from a given --src-dir.

* Tools/Scripts/webkitpy/w3c/test_downloader.py:
(TestDownloader._add_test_suite_paths):
(TestDownloader.copy_tests.should_copy_dir):
(TestDownloader.copy_tests.should_copy_file):
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.should_skip_path):
(TestImporter.find_importable_tests):
(TestImporter.should_skip_file): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: https://commits.webkit.org/267088@main
